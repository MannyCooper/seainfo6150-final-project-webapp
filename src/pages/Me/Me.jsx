import React from 'react'
import styles from './Me.module.css'
import portrait from '../../images/portrait.png'

const Me = () => {
    return (
        <div className={styles.me_container}>
            <h1 className={styles.title}>About<br/>Me<br/><span role="img" aria-label="Me">üßëüèª‚Äçüíª</span></h1>
            <img className={styles.portrait} src={portrait} alt="portrait" />
            <article>
                <h2 className={styles.hi}>
                    <span role="img" aria-label="Hi">üëã </span>
                    Zifan An
                </h2>
                <div className={styles.intro}>
                    <p>Postgrad at the <a className={styles.NEU} href="https://www.northeastern.edu" target="_blank" rel="noopener noreferrer">Northeastern University</a>. Alumni of the <a className={styles.NJUPT} href="http://www.njupt.edu.cn/en/" target="_blank" rel="noopener noreferrer">Nanjing University of Posts and Telecommunications</a>. Technical editors at <a className={styles.SSPAI} href="https://sspai.com" target="_blank" rel="noopener noreferrer">SSPAI</a>.</p>
                    <p>
                        In the past four years, I have achieved strong skills in web development, data science, and machine learning. I'm familiar with Java, Python, and JavaScript. And, of course, the associated frameworks and libraries. You can see the technical stack in more detail below. At the same time, I also published a paper in the field of AI related to paralanguage: Exploration of Acoustic and Lexical Cues for the INTERSPEECH 2020 Computational Paralinguistic Challenge.
                </p>
                    <h2 className={styles.skills}>Technical Skills:</h2>
                    <ul>
                        <li>
                            <p><b>Programming Languages:</b></p>
                            <p>Java, Python, JavaScript, HTML, CSS, Swift, SQL, MATLAB</p>
                        </li>
                        <li>
                            <p><b>Framework/Libraries:</b></p>
                            <p>Vue.js, React.js, Node.js, scikit-learn, TensorFlow, Hadoop, SwiftUl, JavaFX, Scrapy, Git...</p>
                        </li>
                        <li>
                            <p><b>Database:</b></p>
                            <p>MySQL, MongoDB, Redis</p>
                        </li>
                    </ul>
                </div>
            </article>
        </div>
    )
}

export default Me
