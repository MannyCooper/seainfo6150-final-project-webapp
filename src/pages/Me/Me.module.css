.me_container {
    max-width: 1100px;
    margin: 30px auto;
    background-color: white;
    border-radius: 25px;
    padding: 30px;
}

.portrait{
    height: 100px;
    width: 100px;
    border-radius: 99px; 
}

.title{
    font-size: 32px;
    line-height: 1.1;
    font-weight: 700;
    letter-spacing: 0.004em;
    margin-bottom: 24px;
    float: right;
    text-align: right;
}

.hi{
    margin-top: 2rem;
    font-weight: 700;
    font-size: 1.5rem;
    line-height: 2rem;
}

.intro{
    line-height: 1.6;
    margin-top: 3rem;
}

.intro p{
    margin-top: .5rem;
    line-height: 1.6;
}

.NEU {
    color: #d41b2c;
    position: relative;
    font-weight: 500;
}

.NJUPT {
    color: #213f9a;
    position: relative;
    font-weight: 500;
}

.SSPAI {
    color: #da282a;
    position: relative;
    font-weight: 500;
}

.NEU::before,.NJUPT::before,.SSPAI::before {
    content: "";
    position: absolute;
    width: 100%;
    height: 2px;
    bottom: 0;
    left: 0;
    visibility: hidden;
    transform: scaleX(0);
    transition: all 0.3s ease-in-out;
}

.NEU::before{
    background-color: #d41b2c;
}
.NJUPT::before{
    background-color: #213f9a;
}
.SSPAI::before{
    background-color: #da282a;
}

.NEU:hover::before,.NJUPT:hover::before,.SSPAI:hover::before {
    visibility: visible;
    transform: scaleX(1);
}

@media only screen and (max-width: 1380px){
    .me_container{
        max-width: 980px;
    }
}

@media only screen and (max-width: 1200px){
    .me_container{
        width: 692px;
    }
}

@media only screen and (max-width: 768px){
    .me_container{
        max-width: 366px;
        width: 87.5%;
    }
    .intro{
        line-height: 1.6;
        margin-top: 1rem;
    }
}

.skills{
    font-size: 18px;
    font-weight: 600;
    margin-top: 30px;
}

li{
    display: flex;
    justify-content: space-between;
}