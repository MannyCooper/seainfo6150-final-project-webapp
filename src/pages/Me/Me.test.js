 /* global render */
import React from "react";
import Me from "./Me";

describe("Me tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Me />);
    expect(container).toMatchSnapshot();
  });
});
