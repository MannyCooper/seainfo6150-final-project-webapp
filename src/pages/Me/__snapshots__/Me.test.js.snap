// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Me tests renders correctly 1`] = `
<div>
  <div
    class="me_container"
  >
    <h1
      class="title"
    >
      About
      <br />
      Me
      <br />
      <span
        aria-label="Me"
        role="img"
      >
        üßëüèª‚Äçüíª
      </span>
    </h1>
    <img
      alt="portrait"
      class="portrait"
      src="portrait.png"
    />
    <article>
      <h2
        class="hi"
      >
        <span
          aria-label="Hi"
          role="img"
        >
          üëã 
        </span>
        Zifan An
      </h2>
      <div
        class="intro"
      >
        <p>
          Postgrad at the 
          <a
            class="NEU"
            href="https://www.northeastern.edu"
            rel="noopener noreferrer"
            target="_blank"
          >
            Northeastern University
          </a>
          . Alumni of the 
          <a
            class="NJUPT"
            href="http://www.njupt.edu.cn/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Nanjing University of Posts and Telecommunications
          </a>
          . Technical editors at 
          <a
            class="SSPAI"
            href="https://sspai.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            SSPAI
          </a>
          .
        </p>
        <p>
          In the past four years, I have achieved strong skills in web development, data science, and machine learning. I'm familiar with Java, Python, and JavaScript. And, of course, the associated frameworks and libraries. You can see the technical stack in more detail below. At the same time, I also published a paper in the field of AI related to paralanguage: Exploration of Acoustic and Lexical Cues for the INTERSPEECH 2020 Computational Paralinguistic Challenge.
        </p>
        <h2
          class="skills"
        >
          Technical Skills:
        </h2>
        <ul>
          <li>
            <p>
              <b>
                Programming Languages:
              </b>
            </p>
            <p>
              Java, Python, JavaScript, HTML, CSS, Swift, SQL, MATLAB
            </p>
          </li>
          <li>
            <p>
              <b>
                Framework/Libraries:
              </b>
            </p>
            <p>
              Vue.js, React.js, Node.js, scikit-learn, TensorFlow, Hadoop, SwiftUl, JavaFX, Scrapy, Git...
            </p>
          </li>
          <li>
            <p>
              <b>
                Database:
              </b>
            </p>
            <p>
              MySQL, MongoDB, Redis
            </p>
          </li>
        </ul>
      </div>
    </article>
  </div>
</div>
`;
